types/jest/jest-tests.ts

```diff
@@ -572,71 +357,6 @@ jest.spyOn(spiedPromiseTarget, 'resolvesString')
     .mockRejectedValue('value')
     .mockRejectedValueOnce('value');
 
-let spy4: jest.SpyInstance;
-// $ExpectType SpyInstance<string, []>
-spy4 = jest.spyOn(spiedTarget, 'returnsString');
-// compiles because spy4 is declared as jest.SpyInstance<any, any>
-spy4.mockImplementation(() => 1);
-spy4.mockRestore();
-
-let spy5: jest.SpiedFunction<typeof spiedTarget.setValue>;
-
-// $ExpectType SpyInstance<void, [string]> || SpyInstance<void, [value: string]>
-spy5 = jest.spyOn(spiedTarget, 'setValue');
-// @ts-expect-error
-spy5 = jest.spyOn(spiedTarget, 'returnsString');
-
-// $ExpectType SpyInstance<number, []>
-const spy6 = jest.spyOn(spiedTarget2, 'value', 'get');
-// @ts-expect-error
-spy6.mockReturnValue('5');
-
-// $ExpectType SpyInstance<void, [number]>
-jest.spyOn(spiedTarget2, 'value', 'set');
-
-const spyInterfaceImpl: SpyInterface = {};
-// @ts-expect-error
-jest.spyOn(spyInterfaceImpl, 'method', 'get');
-// @ts-expect-error
-jest.spyOn(spyInterfaceImpl, 'prop');
-// $ExpectType SpyInstance<number, []>
-jest.spyOn(spyInterfaceImpl, 'prop', 'get');
-// $ExpectType SpyInstance<void, [boolean]> || SpyInstance<void, [arg1: boolean]>
-jest.spyOn(spyInterfaceImpl, 'method');
```

Comment:
Just to draw attention – some useful tests might be removed accidentally. E.g., this particular test helped me to figure out that Jest’s build-in `spyOn` typings do not handle optional props in interfaces. Fixed in https://github.com/facebook/jest/pull/13247

Reply:
We should ideally keep most of the tests here, at least during migration. There's gonna be a bunch we don't want/need, but nice to see run the tests in the meantime